pkg/opensearch/client/models.go

```diff
@@ -156,6 +173,28 @@ func (f *QueryStringFilter) MarshalJSON() ([]byte, error) {
 	return json.Marshal(root)
 }
 
+// TermsFilter represents a terms filter
+type TermsFilter struct {
+	Key    string
+	Values []string
+}
+
+func (t TermsFilter) MarshalJSON() ([]byte, error) {
+	if len(t.Values) == 1 {
+		return json.Marshal(map[string]map[string]map[string]string{
+			"term": {
+				t.Key: {"value": t.Values[0]},
+			},
+		})
+	} else {
```

Comment:
True, but I feel like adding the `else` explicitly is more readable 🤔

Reply:
This is a short enough function that it doesn't really matter, but generally, the go style convention is to avoid having returns in elses, and `golint` actually will format your code that way (though we don't have `golint` turned on in our repos). Basically the idea for readability is that your functions are always returning at the end with the general behavior as left justified as possible, with exceptions branching off with an `if` (splitting off into sub functions if really necessary). 
(My old mentor's position was that you should never have an `else` in your go code ever, and while I'm not that extreme I do see his point that it ends up with a lot of different behavior paths that are easy to lose bugs in)
(Also If you want to read people talking about it on reddit here's a [link](https://www.reddit.com/r/golang/comments/e0p20u/golang_code_style_return_in_else_clause_what_do/), lol)